connectors/sync_job_runner.py

```diff
@@ -326,14 +326,14 @@ async def _execute_content_sync_job(self, job_type, bulk_options):
         )
 
     async def _sync_done(self, sync_status, sync_error=None):
-        if self.sync_orchestrator is not None and not self.sync_orchestrator.done():
-            await self.sync_orchestrator.cancel()
         if self.job_reporting_task is not None and not self.job_reporting_task.done():
             self.job_reporting_task.cancel()
             try:
                 await self.job_reporting_task
             except asyncio.CancelledError:
                 self.sync_job.log_debug("Job reporting task is stopped.")
+        if self.sync_orchestrator is not None and not self.sync_orchestrator.done():
+            await self.sync_orchestrator.cancel()
```

Comment:
@artem-shelkovnikov this is the change you suggested, and it does seem to get rid of the `task exception was never received` issue on ctrl+c